gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/EncounterMapper.java

```diff
@@ -0,0 +1,327 @@
+        return ehrComposition.getComponent()
+            .stream()
+            .map(RCMRMT030101UK04Component4::getLinkSet)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+    }
+
+    private List<RCMRMT030101UK04CompoundStatement> getCategoryCompoundStatements(RCMRMT030101UK04CompoundStatement
+        topicCompoundStatement) {
+        return topicCompoundStatement.getComponent()
+            .stream()
+            .map(RCMRMT030101UK04Component02::getCompoundStatement)
+            .filter(compoundStatement -> compoundStatement != null && compoundStatement.getClassCode().get(0).equals(CATEGORY_CLASS_CODE))
+            .toList();
+    }
+
+    private List<RCMRMT030101UK04EhrComposition> getEncounterEhrCompositions(RCMRMT030101UK04EhrExtract ehrExtract) {
+        return ehrExtract
+            .getComponent()
+            .stream()
+            .filter(EhrResourceExtractorUtil::hasEhrFolder)
+            .map(RCMRMT030101UK04Component::getEhrFolder)
+            .map(RCMRMT030101UK04EhrFolder::getComponent)
+            .flatMap(List::stream)
+            .filter(EhrResourceExtractorUtil::hasEhrComposition)
+            .map(RCMRMT030101UK04Component3::getEhrComposition)
+            .filter(ehrComposition -> isEncounterEhrComposition(ehrComposition.getCode().getCode(), ehrComposition.getComponent()))
+            .toList();
+    }
+
+    private boolean isEncounterEhrComposition(String code, List<RCMRMT030101UK04Component4> component) {
+        return !INVALID_CODES.contains(code) && !component.stream().anyMatch(this::hasSuppressedContent);
+    }
+
+    private boolean hasSuppressedContent(RCMRMT030101UK04Component4 component) {
+        return component.getEhrEmpty() != null || component.getRegistrationStatement() != null;
+    }
+
+    private List<RCMRMT030101UK04CompoundStatement> getTopicCompoundStatements(RCMRMT030101UK04EhrComposition ehrComposition) {
+        return ehrComposition
+            .getComponent()
+            .stream()
+            .map(RCMRMT030101UK04Component4::getCompoundStatement)
+            .filter(compoundStatement -> compoundStatement != null && compoundStatement.getClassCode().get(0).equals(TOPIC_CLASS_CODE))
+            .toList();
+    }
+
+    private Encounter mapToEncounter(RCMRMT030101UK04EhrComposition ehrComposition, Patient patient) {
+        var id = ehrComposition.getId().getRoot();
+
+        var encounter = new Encounter();
+        encounter
+            .setParticipant(getEncounterParticipant(ehrComposition.getAuthor(), ehrComposition.getParticipant2()))
+            .setStatus(EncounterStatus.FINISHED)
+            .setSubject(new Reference(patient))
+            .setType(getEncounterType(ehrComposition.getCode()))
+            .setPeriod(getPeriod(ehrComposition.getEffectiveTime(), ehrComposition.getAvailabilityTime()))
+            .setIdentifier(getEncounterIdentifier(id))
+            .setMeta(getEncounterMeta())
```

Comment:
minor - probably just `getMeta` would do (could drop "encounter" off any method name since we're in the encounter mapper class"